crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -381,6 +457,234 @@ impl Provider for KvRedisProvider {
+        };
+        let mut conn = client.get_connection_manager().await.map_err(|e| {
+            error!("Failed to get async connection: {e}");
+            anyhow::anyhow!("Failed to get async connection: {}", e)
+        })?;
+
+        let component_id: Arc<str> = target_id.into();
+        let wrpc = get_connection()
+            .get_wrpc_client(&component_id)
+            .await
+            .context("failed to construct wRPC client")?;
+        if interfaces.contains(&"watcher".to_string()) {
+            let config_response: Vec<String> = redis::cmd("CONFIG")
+                .arg("GET")
+                .arg("notify-keyspace-events")
+                .query_async(&mut conn)
+                .await
+                .map_err(|e| {
+                    error!("Failed to get keyspace notifications config: {e}");
+                    anyhow::anyhow!("Failed to get keyspace notifications config: {}", e)
+                })?;
+
+            let current_config = config_response.get(1).ok_or_else(|| {
+                error!("Unexpected response format from Redis CONFIG GET");
+                anyhow::anyhow!("Unexpected response format from Redis CONFIG GET")
+            })?;
+
+            if !current_config.contains('K')
+                || !current_config.contains('E')
+                || !current_config.contains('A')
+            {
+                error!("Redis keyspace-notifications not properly configured");
+                return Err(anyhow::anyhow!(
+                    "Redis keyspace-notifications not properly configured! \
+                        Expected 'KEA' in settings, but got '{}'. \
+                        Please run: CONFIG SET notify-keyspace-events KEA",
+                    current_config
+                ));
+            }
+
+            let wrpc = Arc::new(wrpc);
+            let wrpc_for_task = wrpc.clone();
+
+            let mut watched_keys = self.watched_keys.write().await;
+
+            let config: HashMap<UniCase<&str>, &String> = config
+                .iter()
+                .map(|(k, v)| (UniCase::new(k.as_str()), v))
+                .collect();
+
+            // Look for watch configuration in the format "watch: SET@key:value,DEL@key"
+            if let Some(watch_config) = config.get(&UniCase::new("watch")) {
+                for watch_entry in watch_config.split(',') {
+                    let watch_entry = watch_entry.trim();
+                    if watch_entry.is_empty() {
+                        continue;
+                    }
+                    let parts: Vec<&str> = watch_entry.split('@').collect();
+                    if parts.len() != 2 {
+                        error!("Invalid watch entry format: {watch_entry}. Expected FORMAT@KEY[:VALUE]");
+                    }
+
+                    let operation = parts[0].trim().to_uppercase();
+                    let key_value = parts[1].trim();
+
+                    match operation.as_str() {
+                        "SET" | "set" => {
```

Comment:
Nit: You `to_uppercase` this above, so you should only need to match on `SET` or `DEL`